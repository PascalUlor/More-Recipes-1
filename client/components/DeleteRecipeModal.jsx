import React from 'react';
import PropTypes from 'prop-types';


/**
 * @description displays delete recipe modal component
 *
 * @param { function } handleDelete - handles deleting of recipes
 *
 * @returns { jsx } jsx - renders delete recipe modal
 */
const DeleteRecipeModal = ({ handleDelete }) => (
  <div className="modal fade" id="deleteRecipeModal" tabIndex="-1"
    role="dialog" aria-labelledby="deleteRecipe" aria-hidden="true">
    <div className="modal-dialog" role="document">
      <div className="modal-content">
        <div className="modal-header">
          <span className="navbar-brand text-gray-dark">
            <img src="/images/logo.png" width="45" height="32"
              className="d-inline-block align-center" alt=" Make Recipe Logo"/>
            <span id="site-name">More Recipes</span>
          </span>
          <button type="button" id="close" className="close"
            data-dismiss="modal" aria-label="Close">
            <span aria-hidden="true"><i className="fa fa-times-circle"
              aria-hidden="true"></i></span>
          </button>
        </div>
        <div className="modal-body text-muted text-left">
          <div className="container-fluid">
            Are you sure of deleting this recipe?
          </div>
        </div>
        <div className="modal-footer">
          <button type="button" className="btn btn-secondary"
            data-dismiss="modal">Cancel
          </button>
          <button type="button" id="delete-recipe" onClick={handleDelete}
            className="btn btn-danger">Delete
          </button>
        </div>
      </div>
    </div>
  </div>
);

DeleteRecipeModal.propTypes = {
  handleDelete: PropTypes.func.isRequired
};

export default DeleteRecipeModal;
