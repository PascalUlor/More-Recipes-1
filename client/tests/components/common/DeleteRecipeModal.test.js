/* eslint-disable */
import React from 'react';
import toJson from 'enzyme-to-json';
import DeleteRecipeModal from '../../../components/DeleteRecipeModal.jsx';


describe('<DeleteRecipeModal/>', () => {
  const props = {
    handleDelete: () => {}
  };
  it('renders delete recipe modal component without crashing', () => {
    const shallowWrapper = shallow(<DeleteRecipeModal {...props}/>);
    expect(toJson(shallowWrapper)).toMatchSnapshot();
  });
});
