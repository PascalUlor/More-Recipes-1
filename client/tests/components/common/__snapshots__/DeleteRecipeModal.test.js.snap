// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DeleteRecipeModal/> renders delete recipe modal component without crashing 1`] = `
<div
  aria-hidden="true"
  aria-labelledby="deleteRecipe"
  className="modal fade"
  id="deleteRecipeModal"
  role="dialog"
  tabIndex="-1"
>
  <div
    className="modal-dialog"
    role="document"
  >
    <div
      className="modal-content"
    >
      <div
        className="modal-header"
      >
        <span
          className="navbar-brand text-gray-dark"
        >
          <img
            alt=" Make Recipe Logo"
            className="d-inline-block align-center"
            height="32"
            src="/images/logo.png"
            width="45"
          />
          <span
            id="site-name"
          >
            More Recipes
          </span>
        </span>
        <button
          aria-label="Close"
          className="close"
          data-dismiss="modal"
          id="close"
          type="button"
        >
          <span
            aria-hidden="true"
          >
            <i
              aria-hidden="true"
              className="fa fa-times-circle"
            />
          </span>
        </button>
      </div>
      <div
        className="modal-body text-muted text-left"
      >
        <div
          className="container-fluid"
        >
          Are you sure of deleting this recipe?
        </div>
      </div>
      <div
        className="modal-footer"
      >
        <button
          className="btn btn-secondary"
          data-dismiss="modal"
          type="button"
        >
          Cancel
        </button>
        <button
          className="btn btn-danger"
          onClick={[Function]}
          type="button"
        >
          Delete
        </button>
      </div>
    </div>
  </div>
</div>
`;
